plugins/PrivacyManager/vue/src/ReportDeletionSettings/ReportDeletionSettings.store.ts

```diff
@@ -0,0 +1,147 @@
+/*!
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+import {
+  reactive,
+  computed,
+  readonly,
+} from 'vue';
+import {
+  AjaxHelper,
+  NotificationsStore,
+  translate,
+} from 'CoreHome';
+
+export interface ReportDeletionSettings {
+  enableDeleteLogs?: boolean;
+  enableDeleteReports?: boolean;
+  deleteLogsOlderThan?: string;
+  deleteReportsOlderThan?: string;
+  keepBasic?: string;
+  keepDay?: string;
+  keepWeek?: string;
+  keepMonth?: string;
+  keepYear?: string;
+  keepRange?: string;
+  keepSegments?: string;
+  deleteLowestInterval?: string;
+}
+
+interface ReportDeletionSettingsStoreState {
+  settings: ReportDeletionSettings;
+  showEstimate: boolean;
+  loadingEstimation: boolean;
+  estimation: string;
+  isModified: boolean;
+}
+
+class ReportDeletionSettingsStore {
+  private privateState = reactive<ReportDeletionSettingsStoreState>({
+    settings: {},
+    showEstimate: false,
+    loadingEstimation: false,
+    estimation: '',
+    isModified: false,
+  });
+
+  readonly state = computed(() => readonly(this.privateState));
+
+  readonly enableDeleteReports = computed(() => this.state.value.settings.enableDeleteReports);
+
+  readonly enableDeleteLogs = computed(() => this.state.value.settings.enableDeleteLogs);
+
+  private currentRequest?: AbortController;
+
+  updateSettings(settings: ReportDeletionSettings) {
+    this.initSettings(settings);
+    this.privateState.isModified = true;
+  }
+
+  initSettings(settings: ReportDeletionSettings) {
+    this.privateState.settings = { ...this.privateState.settings, ...settings };
+    this.reloadDbStats();
+  }
+
+  savePurgeDataSettings(apiMethod: string, settings: ReportDeletionSettings) {
+    this.privateState.isModified = false;
+    return AjaxHelper.post(
+      {
+        module: 'API',
+        method: apiMethod,
+      },
+      {
+        ...settings,
+        enableDeleteLogs: settings.enableDeleteLogs ? '1' : '0',
+        enableDeleteReports: settings.enableDeleteReports ? '1' : '0',
```

Comment:
This doesn't seem to work properly. For me the request to calculate the stats always only include `enableDeleteLogs` and  `enableDeleteReports`, all other settings are missing and so the calculated value isn't correct.

Reply:
@sgiehl thanks for catching this :+1:, should be fixed. Also tested the manual estimate feature.